hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/InterSCMGrpcClient.java

```diff
@@ -52,13 +52,15 @@
   private final InterSCMProtocolServiceGrpc.InterSCMProtocolServiceStub
       client;
 
-  private final long timeout;
-
-  public InterSCMGrpcClient(final String host, final ConfigurationSource conf) {
-    Preconditions.checkNotNull(conf);
-    int port = conf.getInt(ScmConfigKeys.OZONE_SCM_GRPC_PORT_KEY,
-        ScmConfigKeys.OZONE_SCM_GRPC_PORT_DEFAULT);
-    timeout =
+  public InterSCMGrpcClient(final String host, final int leaderPort,
+      final ConfigurationSource conf) {
+    // if the leader grpc port details are not setup in the peer Map,
+    // fall back to default grpc port.
+    final int port = leaderPort == 0 ?
```

Comment:
Do we need this, because grpcPort is set in SCMNodeDetails during build?

Reply:
This is required for tests. With Multiple SCM setups, grpc ports for individual SCMs will be different. But, grpc client needs to connect leader grpc port which it does not know from the leaderInfo from Ratis. In actual production clusters, all grpc servers on all nodes will bind to the same grpc port which is not possible in MiniOzoneHAcluster.

Reply:
If each SCM is having different Grpc port, it can get leader grpc port from nodeDetails Right?
If we have already set grpc port in tests.

Reply:
fixed the tests. Thanks.

Reply:
The port here follows the HA style config, since it is in `OZONE_SCM_GRPC_PORT_KEY` of `nodeSpecificConfigKeys`, thus it can be handled in MiniOzoneCluster, right ?